enterprise/server/remote_execution/runner/runner.go

```diff
@@ -381,39 +382,72 @@ func (r *commandRunner) Run(ctx context.Context) *interfaces.CommandResult {
 
 	execResult := r.Container.Exec(ctx, command, &commandutil.Stdio{})
 
+	r.checkMaxResourceUtilization(ctx, execResult.UsageStats)
```

Comment:
This has a side effect of setting `doNotRecycle` which affects control flow a bunch, so it may be worth calling out here.

```suggestion
	// Set `doNotRecycle` if the runner is close to its resource limits.
	r.checkMaxResourceUtilization(ctx, execResult.UsageStats)
```

Reply:
Moved the side effect out of the helper function, so the behavior is not as hidden